win32ss/user/user32/misc/logon.c

```diff
@@ -83,6 +84,69 @@ Logon(BOOL IsLogon)
                          sizeof(*LogonRequest));
 }
 
+/* Win: LoadPreloadKeyboardLayouts */
+BOOL IntLoadPreloadKeyboardLayouts(VOID)
+{
+    UINT nNumber, uFlags;
+    DWORD cbValue, dwType;
+    WCHAR szNumber[32], szValue[9];
+    HKEY hPreloadKey;
+    BOOL ret = FALSE;
+    HKL hKL, hDefaultKL = NULL;
+
+    if (RegOpenKeyW(HKEY_CURRENT_USER,
+                    L"Keyboard Layout\\Preload",
+                    &hPreloadKey) != ERROR_SUCCESS)
+    {
+        return FALSE;
+    }
+
+    for (nNumber = 1; nNumber <= 1000; ++nNumber)
+    {
+        StringCchPrintfW(szNumber, _countof(szNumber), L"%u", nNumber);
+
+        cbValue = sizeof(szValue);
+        if (RegQueryValueExW(hPreloadKey,
+                             szNumber,
+                             NULL,
+                             &dwType,
+                             (LPBYTE)szValue,
+                             &cbValue) != ERROR_SUCCESS)
+        {
+            break;
+        }
+
+        if (dwType != REG_SZ)
+            continue;
+
+        if (nNumber == 1)
```

Comment:
Having a comment explaining what this particular case is and why we need different flags here, would be useful.

Reply:
Added a comment.